packages/yoroi-extension/app/api/localStorage/index.js

```diff
@@ -114,15 +114,21 @@ export default class LocalStorageApi {
 
   // ========== Select Wallet ========== //
 
-  getSelectedWalletId: void => number | null = () => {
-    const id = localStorage.getItem(storageKeys.SELECTED_WALLET);
-    if (!id) return null;
+  getSelectedWalletId: void => Promise<number | null> = async () => {
+    let id = await getLocalItem(storageKeys.SELECTED_WALLET);
+    // previously it was stored in window.localStorage, which is not accessible in the mv3 service worker
+    if (!id) {
+      id = window?.localStorage.getItem(storageKeys.SELECTED_WALLET);
+    }
```

Comment:
Must either not do this extra check in the old storage, or also write the result into the new storage, when it's present.